src/main/java/io/vertx/core/json/Json.java

```diff
@@ -166,7 +167,7 @@ public static String encodePrettily(Object obj) throws EncodeException {
    */
   public static <T> T decodeValue(Buffer buf, Class<T> clazz) throws DecodeException {
     try {
-      return mapper.readValue(new ByteBufInputStream(buf.getByteBuf()), clazz);
+      return mapper.readValue((InputStream) new ByteBufInputStream(buf.getByteBuf()), clazz);
```

Comment:
Why would you want/need to cast this to InputStream. Doesn't ByteBufInputStream IMPLEMENT InputStream?